src/recommenderapp/app.py

```diff
@@ -17,29 +17,127 @@
 #pylint: disable=wrong-import-position
 from src.prediction_scripts.item_based import recommend_for_new_user
 #pylint: enable=wrong-import-position
-
+import db
 
 app = Flask(__name__)
 app.secret_key = "secret key"
 
 cors = CORS(app, resources={r"/*": {"origins": "*"}})
 
+# Connect to the database
+db.connect()
+# Initialize the database for users
+db.mutation_query('''
+        CREATE TABLE IF NOT EXISTS users (
+            email    varchar(500),
+            password    varchar(500)
+        )
+    ''')
+
+def get_user():
+    """
+    Returns the user's email if a user is logged in, otherwise returns none
+    """
+    ret = 'None'
+    if 'email' in session:
+        # An email is in the session
+        ret = session['email']
+    return ret
 
 @app.route("/")
 def landing_page():
     """
-    Renders the landing page.
+    Renders the landing page with the user.
     """
-    return render_template("landing_page.html")
+    return render_template("landing_page.html", user=get_user())
 
+@app.route("/login")
+def login_page():
+    """
+    Renders the login page.
+    """
+    return render_template("login.html", user=get_user())
+
+@app.route("/signup")
+def signup_page():
+    """
+    Renders the signup page.
+    """
+    return render_template("signup.html", user=get_user())
 
 @app.route("/search_page")
 def search_page():
     """
     Renders the search page.
     """
-    return render_template("search_page.html")
+    return render_template("search_page.html", user=get_user())
 
+@app.route('/processSignup', methods = ["POST","GET"])
+def processSignup():
+    """
+    Queries the database to see if the user already exists. If so it lets the javascript know
+    If not, the new user is added to the database
+    """
+    ret = {}
+    # Below is how we create a dictionary of the data sent from the input form
+    form_fields = dict((key, request.form.getlist(key)[0]) for key in list(request.form.keys()))
+    print(form_fields)
+
+    # query to see if user in the database already
+    user_query = f"SELECT * FROM users where email=?"
+    result = db.select_query(user_query, (form_fields["email"],))
+    print(result)
+    print('blehhhhh')
```

Comment:
Please remove the logging/debug statements if not needed.

Reply:
Fixed